### PR TITLE
copr: vectorize Log1Arg Log2Args Log2 Log10

### PR DESCRIPTION
**Pull request by Fullstop000**
PCP https://github.com/tikv/tikv/issues/5751
Signed-off-by: Fullstop000 <fullstop1005@gmail.com>

###  What have you changed?

Please explain in detail what the changes are in this PR and why they are needed:

- Vectorize Log1Arg Log2Args Log2 Log10

###  What is the type of the changes?

Pick one of the following and delete the others:

- Improvement 

###  How is the PR tested?
Please select the tests that you ran to verify your changes:
- Unit test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No

###  Does this PR affect `tidb-ansible`?

No

### CODE REVIEW

**Comment by breeswish** [comment]
Friendly hint.. You will only get score once if you introduce multiple functions in one PR. It is a limitation of the PCP bot. You may want to separate it into multiple PRs to maximize the score.

**Comment by Fullstop000** [comment]
@breeswish I know that rule about the points. But I still want to finish all of them since they're similar to each other. Let's be pragmatism but not utilitarian 😝.

File: components/tidb_query/src/rpn_expr/impl_math.rs [hunk 1]
```diff
@@ -188,6 +243,80 @@ mod tests {
 
     use crate::rpn_expr::types::test_util::RpnFnScalarEvaluator;
 
+    #[test]
+    fn test_log_1_arg() {
+        let test_cases = vec![
+            (std::f64::consts::E, Some(Real::from(1.0_f64))),
+            (100.0, Some(Real::from(4.605170185988092_f64))),
+            (-1.0, None),
+            (0.0, None),
```
Comment (breeswish) [inline components/tidb_query/src/rpn_expr/impl_math.rs]: Missing input = None case

**Comment by Fullstop000** [comment]
@breeswish It should be tidb=pr/13359 I think 😂

**Comment by Fullstop000** [comment]
It's strange that CI tells TiDB internal error in `Sqrt` and `Sign`  🤔️

**Comment by Fullstop000** [comment]
@andylokandy PTAL. Thanks!

**Comment by Fullstop000** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/9 tidb=pr/13359

**Comment by Fullstop000** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/9 tidb=pr/13359

File: components/tidb_query/src/rpn_expr/impl_math.rs [hunk 2]
```diff
@@ -5,6 +5,61 @@ use crate::codec::{self, Error};
 use crate::expr::EvalContext;
 use crate::Result;
 
+#[inline]
+#[rpn_fn]
+pub fn log_1_arg(arg: &Option<Real>) -> Result<Option<Real>> {
+    Ok(arg.and_then(|n| {
+        let res = n.ln();
+        if res.is_finite() {
```
Comment (iosmanthus) [inline components/tidb_query/src/rpn_expr/impl_math.rs]: Seems `if res.is_finite()` is repeated lots of times, would you mind extract this logic?

**Comment by Fullstop000** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/9 tidb=pr/13359

**Comment by Fullstop000** [comment]
@sre-bot /test

**Comment by breeswish** [comment]
@Fullstop000 Please fix conflicts.

**Pull request merged**